pennylane/ops/qubit/qchem_ops.py

```diff
@@ -415,12 +436,29 @@ class DoubleExcitationPlus(Operation):
     num_wires = 4
     grad_method = "A"
 
-    G = -1 * np.eye(16, dtype=np.complex64)
-    G[3, 3] = 0
-    G[12, 12] = 0
-    G[3, 12] = -1j  # 3 (dec) = 0011 (bin)
-    G[12, 3] = 1j  # 12 (dec) = 1100 (bin)
-    generator = [G, -1 / 2]
+    def generator(self):
+        rows = [0, 1, 2, 4, 5, 6, 7, 8, 9, 10, 11, 13, 14, 15, 3, 12]
+        cols = [0, 1, 2, 4, 5, 6, 7, 8, 9, 10, 11, 13, 14, 15, 12, 3]
+        data = [
+            0.5,
+            0.5,
+            0.5,
+            0.5,
+            0.5,
+            0.5,
+            0.5,
+            0.5,
+            0.5,
+            0.5,
+            0.5,
+            0.5,
+            0.5,
+            0.5,
+            0.5j,
+            -0.5j,
+        ]
+        H = coo_matrix((data, (rows, cols)), shape=(16, 16))
+        return qml.SparseHamiltonian(H, wires=self.wires)
```

Comment:
this generator had 16 terms, so I thought maybe better to use `SparseHamiltonian`?

Reply:
Wouldn't 
```python
G = -1 * np.eye(16, dtype=np.complex64)
G[3, 3] = G[12, 12] = 0
G[3, 12] = -1j  # 3 (dec) = 0011 (bin)
G[12, 3] = 1j  # 12 (dec) = 1100 (bin)
H = sp.sparse.coo_matrix(G)
return qml.SparseHamiltonian(H, wires=self.wires)
```
be a bit more convenient to read? As the dense matrix size is fixed, it should be negligible to construct it.

Reply:
Yes, I took a detour here! Will update this.
```suggestion
    def generator(self):
        G = -1 * np.eye(16, dtype=np.complex64)
        G[3, 3] = G[12, 12] = 0
        G[3, 12] = -1j # 3 (dec) = 0011 (bin)
        G[12, 3] = 1j # 12 (dec) = 1100 (bin)
        H = coo_matrix(G)
        return qml.SparseHamiltonian(H, wires=self.wires)
```